File: backend/src/main/java/com/productivity/dto/ApiResponse.java
package com.productivity.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import java.time.LocalDateTime;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class ApiResponse<T> {
    private boolean success;
    private String message;
    private T data;
    private String timestamp;
    private Integer totalPages;
    private Long totalElements;

    private ApiResponse() {
        this.timestamp = LocalDateTime.now().toString();
    }

    public static <T> ApiResponse<T> success(T data) {
        ApiResponse<T> response = new ApiResponse<>();
        response.success = true;
        response.data = data;
        return response;
    }

    public static <T> ApiResponse<T> success(T data, String message) {
        ApiResponse<T> response = success(data);
        response.message = message;
        return response;
    }

    public static <T> ApiResponse<T> paginated(T data, int totalPages, long totalElements) {
        ApiResponse<T> response = success(data);
        response.totalPages = totalPages;
        response.totalElements = totalElements;
        return response;
    }

    public static <T> ApiResponse<T> error(String message) {
        ApiResponse<T> response = new ApiResponse<>();
        response.success = false;
        response.message = message;
        return response;
    }

    public boolean isSuccess() { return success; }
    public String getMessage() { return message; }
    public T getData() { return data; }
    public String getTimestamp() { return timestamp; }
    public Integer getTotalPages() { return totalPages; }
    public Long getTotalElements() { return totalElements; }
}

File: backend-node/src/utils/apiResponse.js
'use strict';

function success(res, data, message, statusCode = 200) {
  return res.status(statusCode).json({
    success: true,
    message: message || undefined,
    data,
    timestamp: new Date().toISOString(),
  });
}

function paginated(res, data, totalPages, totalElements) {
  return res.status(200).json({
    success: true,
    data,
    totalPages,
    totalElements,
    timestamp: new Date().toISOString(),
  });
}

function error(res, message, statusCode = 400) {
  return res.status(statusCode).json({
    success: false,
    message,
    timestamp: new Date().toISOString(),
  });
}

module.exports = { success, paginated, error };
